src/shm_manager.h

```diff
@@ -56,7 +56,7 @@ struct AllocatedSharedMemory {
 
 struct AllocatedShmOwnership {
   uint32_t ref_count_;
-};
+} __attribute__((aligned(16)));
```

Comment:
If it is needed, should the alignment not be 4 since uint32_t is always 4 bytes on both x86 and ARM?

Reply:
Alignment affects the memory address that the object is placed on, it isn't necessarily related to the size of the object.

Reply:
In this particular case, I wanted to size of the object to be 16 bytes so that when the data that is allocated after this object is 16bytes aligned if the initial was aligned.

Reply:
@CoderHam  The natural alignment of struct with uint32_t member will be 4bytes. For some reason storing the ref_count_ in a 16 byte aligned memory is working out for the ARM machines. I would also like to know why that would be needed? The reason should be properly documented in the code.

Reply:
I added a comment. Let me know if it is still not clear.